### PR TITLE
MINOR: Update doc for 3.1

### PR DESCRIPTION
**Pull request by dajac**
* Update main version of main documentation and add link to previous version;
* Update quick start guide (links do not work yet obviously);
* Add upgrade section.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by showuon — COMMENTED** [review]
@dajac , I believe there is one place in doc that needed to be updated, too. Just not sure if you want to include in this PR, or have a separate one.
In `docs/documentation.html`, you should also bump the version to 3.1, and add a `Prior releases` link, as shown in the documentation page:

image

Thanks.

**Comment by dajac** [comment]
@showuon Thanks, good catch.

File: docs/upgrade.html [hunk 1]
```diff
@@ -19,6 +19,48 @@
...
+    </li>
+    <li>Once the cluster's behavior and performance has been verified, bump the protocol version by editing
+        <code>inter.broker.protocol.version</code> and setting it to <code>3.1</code>.
+    </li>
+    <li>Restart the brokers one by one for the new protocol version to take effect. Once the brokers begin using the latest
+        protocol version, it will no longer be possible to downgrade the cluster to an older version.
+    </li>
+    <li>If you have overridden the message format version as instructed above, then you need to do one more rolling restart to
+        upgrade it to its latest version. Once all (or most) consumers have been upgraded to 0.11.0 or later,
+        change log.message.format.version to 2.8 on each broker and restart them one by one. Note that the older Scala clients,
```
Comment (omkreddy) [inline docs/upgrade.html]: => change log.message.format.version to `3.1`

**Review by omkreddy — APPROVED** [review]
not related to this PR, looks like we are missing upgrade instructions to 3.0.0 release in kafka docs and kafka-site. Can we also add those here.

**Comment by dajac** [comment]
@omkreddy Thanks for the review. I have addressed your comments. Could you have another look?

File: docs/upgrade.html [hunk 2]
```diff
@@ -36,8 +78,52 @@ <h5><a id="upgrade_310_notable" href="#upgrade_310_notable">Notable changes in 3
         Those applications will need to go through a bridge release, by first upgrading to a version between 2.4 - 3.1 and setting the <code>upgrade.from</code> config,
         then removing that config and upgrading to the final version above 3.1. See <a href="https://issues.apache.org/jira/browse/KAFKA-8575">KAFKA-8575</a>
         for more details.</li>
+    <li>IBP 3.1 introduces topic IDs to FetchRequest as a part of
+        <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-516%3A+Topic+Identifiers">KIP-516</a>.</li>
 </ul>
 
+<h4><a id="upgrade_3_0_0" href="#upgrade_3_0_0">Upgrading to 3.0.0 from any version 0.8.x through 3.0.x</a></h4>
```
Comment (omkreddy) [inline docs/upgrade.html]: => through `2.8.x`

**Review by omkreddy — APPROVED** [review]
@dajac Thanks for the update. left minor comment. Pls feel free to merge the PR after addressing the comment.

**Pull request merged**